packages/core/src/initHooks.ts

```diff
@@ -22,6 +25,75 @@ import type {
   OnExitFn,
 } from './types';
 
+export function createEnvironmentAsyncHook<
+  Callback extends (...args: any[]) => any,
+>(): EnvironmentAsyncHook<Callback> {
+  type Hook = {
+    environmentMeta?: EnvironmentMeta;
+    handler: Callback;
+  };
+  const preGroup: Hook[] = [];
+  const postGroup: Hook[] = [];
+  const defaultGroup: Hook[] = [];
+
+  const tap =
+    (environmentMeta?: EnvironmentMeta) =>
+    (cb: Callback | HookDescriptor<Callback>) => {
+      if (isFunction(cb)) {
+        defaultGroup.push({
+          environmentMeta,
+          handler: cb,
+        });
+      } else if (cb.order === 'pre') {
+        preGroup.push({
+          environmentMeta,
+          handler: cb.handler,
+        });
+      } else if (cb.order === 'post') {
+        postGroup.push({
+          environmentMeta,
+          handler: cb.handler,
+        });
+      } else {
+        defaultGroup.push({
+          environmentMeta,
+          handler: cb.handler,
+        });
+      }
+    };
+
+  const call =
+    (environment?: string) =>
```

Comment:
Why does `tap` accept `environmentMeta?: EnvironmentMeta`, but `call` accepts `environment?: string`, can we make them consistent.